src/Symfony/Bundle/SecurityBundle/Command/SecurityRoleHierarchyDumpCommand.php

```diff
@@ -0,0 +1,94 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Security\Core\Dumper\MermaidDirectionEnum;
+use Symfony\Component\Security\Core\Dumper\MermaidDumper;
+use Symfony\Component\Security\Core\Role\RoleHierarchyInterface;
+
+/**
+ * Command to dump the role hierarchy as a Mermaid flowchart.
+ *
+ * @author Damien Fernandes <damien.fernandes24@gmail.com>
+ */
+#[AsCommand(name: 'debug:security:role-hierarchy', description: 'Dump the role hierarchy as a Mermaid flowchart')]
+class SecurityRoleHierarchyDumpCommand extends Command
+{
+    public function __construct(
+        private readonly RoleHierarchyInterface $roleHierarchy,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDefinition([
+                new InputOption(
+                    name: 'direction',
+                    shortcut: 'd',
+                    mode: InputOption::VALUE_REQUIRED,
+                    description: 'The direction of the flowchart ['.implode('|', $this->getAvailableDirections()).']',
+                    default: MermaidDirectionEnum::TOP_TO_BOTTOM->value,
+                    suggestedValues: $this->getAvailableDirections()
+                ),
+            ])
+            ->setHelp(<<<'USAGE'
+                The <info>%command.name%</info> command dumps the role hierarchy in Mermaid format.
+
+                <info>Mermaid</info>: %command.full_name% > roles.mmd
+                <info>Mermaid with direction</info>: %command.full_name% --direction=BT > roles.mmd
+                USAGE
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if (null === $this->roleHierarchy) {
+            $io->getErrorStyle()->writeln('<comment>No role hierarchy is configured.</comment>');
+
+            return Command::SUCCESS;
+        }
+
+        $direction = $input->getOption('direction');
+
+        if (!MermaidDirectionEnum::tryFrom($direction)) {
+            $io->getErrorStyle()->writeln('<error>Invalid direction. Available options: '.implode('|', $this->getAvailableDirections()).'</error>');
+
+            return Command::FAILURE;
+        }
+
+        $dumper = new MermaidDumper();
+        $mermaidOutput = $dumper->dump($this->roleHierarchy, MermaidDirectionEnum::from($direction));
+
+        $output->writeln($mermaidOutput);
```

Comment:
this should use `\Symfony\Component\Console\Output\OutputInterface::OUTPUT_RAW` in the options when writing, to avoid resolving formatting styles of the console component in the mermaid output (potentially breaking the mermaid output)